mongodbatlas/resource_mongodbatlas_cluster.go

```diff
@@ -502,36 +502,38 @@ func resourceMongoDBAtlasClusterCreate(d *schema.ResourceData, meta interface{})
 
 	var containerId string
 	if providerName == "AZURE" {
-		if v, ok := d.GetOk("container_id"); ok {
-			containerId = v.(string)
-			_, res, err := conn.Containers.Get(context.Background(), projectID, containerId)
+		options := &matlas.ContainersListOptions{
+			ProviderName: providerName,
+		}
+		containersOriginal, _, _ := conn.Containers.List(context.Background(), projectID, options)
+		if len(containersOriginal) == 0 {
+			stateConf := &resource.StateChangeConf{
+				Pending:    []string{"provisioned_container"},
+				Target:     []string{"different"},
+				Refresh:    resourceListNetworkContainerRefreshFunc(d, conn, containersOriginal),
+				Timeout:    1 * time.Hour,
+				MinTimeout: 10 * time.Second,
+				Delay:      2 * time.Minute,
+			}
+			// Wait, catching any errors
+			containersResp, err := stateConf.WaitForState()
```

Comment:
Also, could you remove the refresh function we don't need, you can use `containerOriginal` to get the `containerID`